netCDF4/_netCDF4.pyx

```diff
@@ -4015,7 +4015,7 @@ rename a `netCDF4.Variable` attribute named `oldname` to `newname`."""
                 # for string vars, if data is not an array
                 # assume it is a python string and raise an error
                 # if it is an array, but not an object array.
-                if not hasattr(data,'ndim'):
+                if not isinstance(data, numpy.ndarray):
```

Comment:
For vlen variables, I still think it's necessary to require that the object is a numpy object array.